test/powershell/Modules/Microsoft.PowerShell.Utility/xml.tests.ps1

```diff
@@ -79,12 +91,16 @@
         }
 
         It "Returns error for invalid xmlnamespace" {
+            $err = $null
             [xml]$xml = "<a xmlns='bar'><b xmlns:b='foo'>hello</b><c>world</c></a>"
-            { Select-Xml -Xml $xml -XPath foo -Namespace @{c=$null} | ShouldBeErrorId "PrefixError,Microsoft.PowerShell.Commands.SelectXmlCommand" }
+            Select-Xml -Xml $xml -XPath foo -Namespace @{c=$null} -ErrorVariable err
+            $err.FullyQualifiedErrorId | Should Be 'PrefixError,Microsoft.PowerShell.Commands.SelectXmlCommand'
         }
 
         It "Returns error for invalid content" {
-            { Select-Xml -Content "hello" -XPath foo | ShouldBeErrorId "InvalidCastToXmlDocument,Microsoft.PowerShell.Commands.SelectXmlCommand" }
+            $err = $null
+            Select-Xml -Content "hello" -XPath foo -ErrorVariable err
+            $err.FullyQualifiedErrorId | Should Be 'InvalidCastToXmlDocument,Microsoft.PowerShell.Commands.SelectXmlCommand'
```

Comment:
Use `ShouldBeErrorId`

Reply:
Hmm, the original feedback was 'why was -ErrorAction Stop' being used so I removed it.  To use ShouldBeErrorId, I'll need to put the -ErroraAction back.

Reply:
Use:

```powershell
{ Select-Xml ... -ErrorAction Stop } | ShouldBeErrorId ...
```

Reply:
As discussed offline; I don't think -ErrorAction is appropriate for non-terminating errors.